### PR TITLE
Change winapi,wio to windows-rs

### LINKED ISSUE #893: `windows-rs` Support
It seems Microsoft maintains their windows api with `windows-rs` crate. Is there any plan to migrate to `windows-rs`?

Actually I tried to make a app using `windows-rs` and I'm facing the problem to cast `windows-rs`'s d3d device to winapi's that.

### PR DESCRIPTION
**Pull request by namse**
closes #893 

I changed winapi and wio with windows-rs which Microsoft maintains, unlike winapi and wio.

I could create d3d example with Microsoft's (https://github.com/microsoft/windows-rs/blob/master/crates/samples/windows/direct3d12/src/main.rs)

### CODE REVIEW

File: skia-safe/Cargo.toml [hunk 1]
```diff
@@ -20,8 +26,17 @@ edition = "2021"
 doctest = false
 
 [features]
-default = ["binary-cache", "embed-icudtl"]
-all-linux = ["gl", "egl", "vulkan", "x11", "wayland", "textlayout", "svg", "webp"]
+default = ["binary-cache", "embed-icudtl", "d3d"]
```
[the remark this answers is not in the document]
Comment (namse) [reply skia-safe/Cargo.toml:29]: Oups... ok!

**Review by namse — COMMENTED** [review]
Remove d3d from default features

**Comment by namse** [comment]
I was able to make a example in d3d12 with this pull request changes.

Uploading ScreenRecorder-20231212-211102.938.mp4…

If you want, I would put this code in examples of skia-safe.

**Comment by pragmatrix** [comment]
@namse Very nice, and yes, a d3d example would be a great addition.

**Comment by pragmatrix** [comment]
@namse Regarding d3d, consider https://github.com/rust-skia/rust-skia/discussions/741

**Comment by namse** [comment]
Wait, What?!?!?!?!? it's super shocking me.... 

After some searching, I found that chrome team uses ANGLE for chrome. hmm....

My purpose of this PR is to support my game development in windows with d3d12, but if ANGLE have good performance, I don't need to make it in d3d12.

How do you think about this PR, is it better to complete to merge this PR or drop it?

**Comment by namse** [comment]
Seems they uses ANGLE for now and later they would move to Dawn with skia graphite, is it right?
Dawn uses d3d12 directly so skia team don't have to maintain d3d12 code directly...
https://dawn.googlesource.com/dawn/+/refs/heads/chromium-gpu-experimental/README.md

**Comment by pragmatrix** [comment]
> is it better to complete to merge this PR or drop it?

Based on the assumption that it will take some time until the Ganesh backend will be removed entirely, probably even years, I suggest to finish it.

**Comment by namse** [comment]
Ok, I agree with that. Let me upload examples of d3d12 on this PR.
Could you give me some hint to solve these CI/CD errors?

**Comment by pragmatrix** [comment]
> these CI/CD errors?

To reproduce, use `cargo build --all-targets`. And if you can't resolve them, leave them, I'll fix them later.

**Comment by namse** [comment]
Will update `skia-org` in this week

**Comment by namse** [comment]
@pragmatrix Would you run ci actions? I fixed most of problem, so I guess it will be successful.

One concern is I think it's the breaking change of rust-skia's d3d. User should use windows crate instead of wio or winapi.

**Comment by pragmatrix** [comment]
> One concern is I think it's the breaking change of rust-skia's d3d. User should use windows crate instead of wio or winapi.

Yes, I will put this into the announcement of the next milestone.

**Comment by namse** [comment]
It only fails on rust beta clippy

**Comment by pragmatrix** [comment]
> It only fails on rust beta clippy

Will be covered in #908.

**Comment by namse** [comment]
908 merged to this PR

**Comment by pragmatrix** [comment]
Merging this without an in-depth review and testing. I'll take a closer look as soon I've access to my Windows machine after the holidays.

Thank you a lot!

**Comment by namse** [comment]
Happy holyday~! I always thank you for your contribution :)

**Pull request merged**